File: CRUD Project/BasketballTeamManagement/BasketballTeamManagement/Model/Employee.swift
//
//  Employee.swift
//  BasketballTeamManagement
//
//  Created by Paul Adam on 31.10.2022.
//

import Foundation

enum Occupation: String, CaseIterable, Identifiable {
    var id: Self { self }
    
    case coach = "Coach"
    case player = "Player"
    case fitness_trainer = "Fitness trainer"
}

struct Employee: Hashable {
    let id = UUID()
    var name: String
    var dateOfBirth: Date
    var occupation: Occupation
    var salary: Double
    var experience: Int
    
    static func mock() -> Employee {
        return .init(name: "Paul Adam", dateOfBirth: Calendar.current.date(from: .init(year: 2001, month: 09, day: 24)) ?? .now, occupation: .player, salary: 1200.50, experience: 10)
    }
    
    static func dummyData() -> [Employee] {
        return [.init(name: "Paul", dateOfBirth: .now, occupation: .coach, salary: 123, experience: 5),
                .init(name: "Dan", dateOfBirth: .now, occupation: .player, salary: 30, experience: 7),
                .init(name: "Alex", dateOfBirth: .now, occupation: .fitness_trainer, salary: 35, experience: 4)
        ]
    }
}

File: basketball_team_management_flutter/lib/models/employee.dart
import 'package:uuid/uuid.dart';

class Employee {
    final String id = const Uuid().v4();
    String name;
    int age;
    String occupation;
    int salary;
    int experience;

    Employee(
        {
            required this.name,
            required this.age,
            required this.occupation,
            required this.salary,
            required this.experience
        }
    );

    static Employee dummyData1() {
        return Employee(name: "Paul", age: 21, occupation: "coach", salary: 3000, experience: 12);
    }

    static Employee dummyData2() {
        return Employee(name: "Dan", age: 22, occupation: "coach", salary: 2000, experience: 5);
    }

    static Employee dummyData3() {
        return Employee(name: "Alex", age: 20, occupation: "player", salary: 1200, experience: 4);
    }

    static List<String> dummyDataOccupations() {
        return ["coach", "player", "physical trainer"];
    }
}